### ISSUE TITLE
Fix for CVE-2020-10188

### ISSUE THREAD

**Issue by utkarsh2102** [marado/netkit-telnet-ssl#5]
Hi there,

There's a CVE-2020-10188 reported against netkit-telnet-ssl.
POC is here: https://appgateresearch.blogspot.com/2020/02/bravestarr-fedora-31-netkit-telnetd_28.html

Would it be possible for you to get this fixed? :)

**Comment by marado** [comment]
Hi, thanks for your warning.
I will probably not work on it myself, and most probably a fix will only land here once it has been provided at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=953478 . However, if that changes, I'll be sure to submit the fix in there.

**Comment by lamby** [comment]
> I will probably not work on it myself, and most probably a fix will only land here once it has been provided

Can you elaborate why? I think everyone is waiting for someone else to fix this issue here which seems a little sub-optimal. This question might land as odd but may I ask why you do not fix it yourself as you appear to be the new upstream? :)

**Comment by marado** [comment]
Huh... I did look at the debian issue again, and realized that Salvatore Bonaccorso ( carnil@debian.org ) marked it with a forwarded-to-address to here, which, I guess, is what made you believe that I am "the new upstream". To be clear, I never considered myself as such, and, honestly, I don't think that that package's debian maintainer would consider it either.

I did create this repository as a fork of debian's, not the other way around, with the intent of being able to try to help and submit a few patches to debian (which I have always been considering 'my upstream'). In particular, for my own purposes, I need a package built with debian's #749322 fixed, which, unfortunately, forces me to use my own packages, because that patch was still not accepted on debian (despite it having been provided six years ago).

In other words, I do not feel like 'the new upstream', not at all, and while I have nothing against dedicating time to find fixes and contribute code (on the contrary), I am hardly the most qualified person to do so (I barely know the code involved, tbh), and I am not even a telnetd user.

This being said, I am not saying I will not try to sometime fix this (or other) netkit-telnet bugs, but I really believe you got the wrong idea about my 'involvement' with this package - and it is certainly not something on my list of priorities.

**Comment by lamby** [comment]
Thanks for the clarification. :+1:

**Comment by carnil** [comment]
Hi,

On Sat, Mar 28, 2020 at 10:19:26AM -0700, Marcos Marado wrote:
> Huh... I did look at the debian issue again, and realized that
> Salvatore Bonaccorso <carnil@debian.org>marked it with a
> forwarded-to-address to here, which, I guess, is what made you
> believe that I am "the new upstream".

Actually it was the other way round :)

I have removed the forwarded indication again to avoid confusion.

**Comment by marado** [comment]
Turns out Debian's netkit-telnet had been patched a long time ago already: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=953477#16 .
